openedx/core/djangoapps/content/block_structure/models.py

```diff
@@ -59,6 +61,63 @@ def _bs_model_storage():
     )
 
 
+class CustomizableFileField(models.FileField):
+    """
+    Subclass of FileField that allows custom settings to not
+    be serialized (hard-coded) in migrations. Otherwise,
+    migrations include optional settings for storage (such as
+    the storage class and bucket name); we don't want to
+    create new migration files for each configuration change.
+    """
+    def __init__(self, *args, **kwargs):
+        kwargs.update(dict(
+            upload_to=_path_name,
+            storage=_bs_model_storage(),
+            max_length=500,  # allocate enough for base path + prefix + usage_key + timestamp in filepath
+        ))
+        super(CustomizableFileField, self).__init__(*args, **kwargs)
+
+    def deconstruct(self):
+        name, path, args, kwargs = super(CustomizableFileField, self).deconstruct()
+        del kwargs['upload_to']
+        del kwargs['storage']
+        del kwargs['max_length']
+        return name, path, args, kwargs
+
+
+@contextmanager
+def _storage_error_handling(bs_model, operation, is_read_operation=False):
```

Comment:
Ah, I see. Are we already at a point at which we can effectively cache this forever on the memcached cluster? If so, I think that's as much mitigation as it's practical to do at the moment. S3 is fundamental enough to AWS infrastructure that surviving a total outage of it would probably require moving things across multiple regions or even cloud providers. We do have to get there, but I think that will involve a lot of work at a layer under this (for instance, I know devops is pushing us over to cross-region replication for S3 right now, if they haven't already).